build_tools/github_actions/ci_build_llvm.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+# Copyright 2022 The StableHLO Authors.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file is similar to build_mlir.sh, but passes different flags for
+# cacheing in GitHub Actions to improve build speeds.
+
+if [[ $# -ne 2 ]] ; then
+  echo "Usage: $0 <path/to/llvm> <build_dir>"
+  exit 1
+fi
+
+# LLVM source
+LLVM_SRC_DIR="$1"
+LLVM_BUILD_DIR="$2"
+
+# Check for LLD
+FLAGS=""
+LLD_FLAG="-DLLVM_ENABLE_LLD=ON"
+if command -v lld &> /dev/null
+then
+  echo "lld found, appending flag '$LLD_FLAG'"
+  FLAGS="$LLD_FLAG"
+else
+  echo "lld not found, using default linker"
+fi
+
+# Call cmake
+echo "Using additional flags: '$FLAGS'"
+
+cmake -GNinja \
+  "$FLAGS" \
+  "-B$LLVM_BUILD_DIR" \
+  "-H$LLVM_SRC_DIR/llvm" \
+  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+  -DCMAKE_CXX_COMPILER=clang++ \
+  -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+  -DCMAKE_C_COMPILER=clang \
+  -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+  -DLLVM_BUILD_TOOLS=OFF \
+  -DLLVM_ENABLE_ASSERTIONS=On \
+  -DLLVM_ENABLE_BINDINGS=OFF \
+  -DLLVM_ENABLE_PROJECTS=mlir \
+  -DLLVM_INCLUDE_TESTS=OFF \
+  -DLLVM_INCLUDE_TOOLS=ON \
+  -DLLVM_INSTALL_UTILS=ON \
+  -DLLVM_TARGETS_TO_BUILD="X86;NVPTX;AMDGPU"
```

Comment:
Consider ordering CMake flags similarly to https://github.com/openxla/stablehlo/blob/9ca259d5092e9cf2c1fa0788a470df6a4fc95f0a/build_tools/build_mlir.sh#L40-L52 to make these files easier to compare side-by-side, e.g. 

```
cmake -GNinja \
  "-H$LLVM_SRC_DIR/llvm" \
  "-B$LLVM_BUILD_DIR" \
  -DLLVM_INSTALL_UTILS=ON \
  "$LLVM_ENABLE_LLD" \
  -DLLVM_ENABLE_PROJECTS=mlir \
  -DLLVM_TARGETS_TO_BUILD="X86;NVPTX;AMDGPU" \
  -DLLVM_INCLUDE_TOOLS=ON \
  -DLLVM_ENABLE_BINDINGS=OFF \
  -DLLVM_BUILD_TOOLS=OFF \
  -DLLVM_INCLUDE_TESTS=OFF \
  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
  -DLLVM_ENABLE_ASSERTIONS=On \
  -DCMAKE_CXX_COMPILER=clang++ \
  -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
  -DCMAKE_C_COMPILER=clang \
  -DCMAKE_C_COMPILER_LAUNCHER=ccache
```

Reply:
The flags in this file are in alphabetical order (sorted by vim). Made the suggested change to make these more consistent, since it seems like alphabetical isn't a convention for cmake flags.